inst/include/mode/r/mode.hpp

```diff
@@ -52,6 +52,55 @@ cpp11::sexp stats_array(const std::vector<size_t>& dat,
   return ret;
 }
 
+int r_index_check(int x, int max) {
```

Comment:
these helpers lifted straight from dust. wonder if it's time to put utils in a separate file